nengo/conftest.py

```diff
@@ -419,10 +427,24 @@ def pytest_runtest_setup(item):
 
 def pytest_terminal_summary(terminalreporter):
     reports = terminalreporter.getreports('passed')
-    if not reports or terminalreporter.config.getvalue('compare') is None:
-        return
-    terminalreporter.write_sep("=", "PASSED")
-    for rep in reports:
-        for name, content in rep.sections:
-            terminalreporter.writer.sep("-", name)
-            terminalreporter.writer.line(content)
+    do_compare = terminalreporter.config.getvalue('compare') is not None
+    if reports and do_compare:  # pragma: no cover
+        terminalreporter.write_sep("=", "PASSED")
+        for rep in reports:
+            for name, content in rep.sections:
+                terminalreporter.writer.sep("-", name)
+                terminalreporter.writer.line(content)
+
+    if resource and terminalreporter.config.option.memory:
+        # Calculate memory usage; details at
+        # http://fa.bianp.net/blog/2013/different-ways-to-get-memory-consumption-or-lessons-learned-from-memory_profiler/  # noqa, pylint: disable=line-too-long
+        rusage_denom = 1024.
+        if sys.platform == 'darwin':  # pragma: no cover
+            # ... it seems that in OSX the output is in different units ...
+            rusage_denom = rusage_denom * rusage_denom
+        mem = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss / rusage_denom
+        terminalreporter.write_sep("=",
+                                   "total memory consumed: %.2f MiB" % mem)
```

Comment:
I'm not sure what exactly is being reported here. Is it the maximum memory allocated during the test run, or the total memory allocated throughout the test run (which would probably be more than the maximum, due to garbage collection), or the memory allocated at the end of the rest run?

Reply:
I'm pretty sure that this is the amount of memory currently allocated to the Python process. Because of the way that Python does garbage collection, this should be a rough analogue to the maximum amount used during testing, though it's possible it never needed quite that much memory, only most of it. Python's GC doesn't give up any memory back to the OS, so it won't be less than the maximum used by any one test.

Sidenote, I would be OK removing this fixture from this PR; it was mostly implemented to make sure that the PR was having any effect at all. But now there's a test in here that all signals are the correct dtype, so that's enough to make sure it's having an effect, and if we really want to profile memory usage, we should probably use an actual memory profiler.

Reply:
We check all signals, and I tried to check build outputs (e.g. eval points, encoders) as well, but I've never been certain that we've covered all arrays. It might be good to do some profiling at some point, though I'm fine to leave that as future work. When I was trying to estimate memory savings with `--memory`, using `float32` on the test suite didn't save as much as I might have hoped, and I'd like to get a better idea of true memory savings on a more practical example.

Reply:
I'm fine leaving it or taking it out.  It does seem like we probably won't use it that much, but the code is already written and tested so 🤷‍♂ .

Reply:
Feels like another thing we should separate into a pytest plugin.